### ISSUE TITLE
Uncaught exception when shutting down the server

### ISSUE THREAD

**Issue by hansonw** [atom/ide-java#8]
I always get a big redbox when I close the last Java file (I guess this is coming after the server shuts down?) Maybe it's a MacOS thing, not sure how much we can do about this.

Console with debug output: 

<img width="1100" alt="screen shot 2017-09-11 at 6 14 38 pm" src="https://user-images.githubusercontent.com/577378/30303454-15db536e-971d-11e7-8ab2-57592f79e456.png">

Stack:

```
Uncaught Error: Header must provide a Content-Length property.
/Users/hansonw/atom-languageclient/node_modules/vscode-jsonrpc/lib/messageReader.js:191
Hide Stack Trace
Error: Header must provide a Content-Length property.
    at StreamMessageReader.onData (/Users/hansonw/atom-languageclient/node_modules/vscode-jsonrpc/lib/messageReader.js:191:27)
    at Socket.<anonymous> (/Users/hansonw/atom-languageclient/node_modules/vscode-jsonrpc/lib/messageReader.js:176:19)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at Socket.Readable.push (_stream_readable.js:134:10)
    at Pipe.onread (net.js:551:20)
```

**Comment by damieng** [comment]
Are you able to repro this every time?  Also, what version of Atom are you on? 

Not able to repro this on my mac where I did all the new shutdown testing.

**Comment by hansonw** [comment]
Traced to https://github.com/eclipse/eclipse.jdt.ls/issues/347

**Comment by damieng** [comment]
One option would be to switch to sockets for now.  Involves setting an environment variable as part of the exec args after finding a spare port - the code to do that is in the ide-php package.

**Comment by fbricon** [comment]
It seems that Cmd+click on a Type (eg. java.lang.String) to open it causes the server to shutdown

**Comment by fbricon** [comment]
Server shutdown seems to be caused by leaving an open Java file to open a new one (which causes the server to restart)

**Comment by damieng** [comment]
@fbricon can you open another issue - this one is specifically about an exception during normal shutdown. Thanks!

**Comment by damieng** [comment]
Fixed in 0.6.2 with latest language server.
